src/ucp/core/ucp_mm.c

```diff
@@ -368,7 +368,7 @@ static void ucp_memh_dereg(ucp_context_h context, ucp_mem_h memh,
     }
     memh->md_map &= ~md_map;
 
-    if (!(flags & UCS_RCACHE_MEM_DEREG_KEEP_PINNED) &&
+    if (!(flags & UCS_RCACHE_MEM_DEREG_REPLACE) &&
```

Comment:
now we have rcache callback for merge, maybe we no longer need UCS_RCACHE_MEM_DEREG_REPLACE flag?